OutOfSchool/OutOfSchool.AuthCommon/Services/ProviderAdminService.cs

```diff
@@ -612,6 +709,32 @@ await providerAdminChangesLogService.SaveChangesLogAsync(providerAdmin, userId,
         return result;
     }
 
+    private List<(string Name, string Value)> GetTrackedUserProperties(User user)
+    {
+        if (user != null && changesLogConfig.TrackedProperties.TryGetValue("ProviderAdmin", out var trackedProperties))
```

Comment:
1. (Code style) You could invert your `if` condition and `return` empty list immediately when `user is null` or there is no `"ProviderAdmin"` key or `trackedProperties` is empty. 
For such code construction (`if - return`) there is no need to write `else` keyword explicitely, thus you get less code nesting in your method body and introduce so called "fail fast" approach.

2. (Logic) As I can see `changesLogConfig.TrackedProperties["ProviderAdmin"]` is read each time when `GetTrackedUserProperties` is called, and it might me redundant or even dangerous for multiple calls (i.e. in `UpdateProviderAdminOperation`). For that case you either do same work few times or get different set of properties to be tracked. 
Ofc, you might state that `changesLogConfig.TrackedProperties["ProviderAdmin"]` is never changed for class `ProviderServiceAdmin` lifetime - then why reread it?
Suggestion: read `"ProviderAdmin"` from `changesLogConfig.TrackedProperties` in class ctor.

Reply:
3. (Code style) Additional work to eliminate nested code could be done for nested `if`s inside `for`. It's up to you to decide if it is worth changing the code.

Reply:
Fixed all points. Thank you!